### PR TITLE
Adding trait selector dialog; migration

### PR DESCRIPTION
**Pull request by phildominguez**
Closes https://github.com/WookieeMatt/Essence20/issues/319

In this PR:
- Creating a new dialog for trait selection to replace checkboxes (borrowed from dnd5e)
- For now, trait selection only applies to Armor. There's an edit button on its item sheet to open the new dialog.
- Display the selected traits on the item sheet as chips
- Armor traits field is now an array of strings instead of a boolean map
- Migration to convert existing traits object (or in older cases, a single string) to an array of strings
- Dialog text size is small for some reason, open to ideas

Testing:
- 🚨 PLEASE TEST ON AN E20 WORLD YOU DON'T CARE ABOUT BREAKING 🚨
- In `beta`, create some armor with different traits selected
- Pull down this branch, `trait-selector`
- Be a pal and open devtools by pressing F12 before launching the test world. If something goes wrong, the logs could be helpful.
- Launch the test world
- You should see messages about the migration starting and eventually completing
- Armor traits should display as chips on the armor sheet
- Edit button should open the trait selector dialog
- Bonus: You can checkout a commit OLDER than `7465fdd` to create an armor with a single trait, and ensure that migration works too

### CODE REVIEW

File: module/helpers/traits.mjs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  event.preventDefault();
+  const a = event.currentTarget;
+  const options = {
+    name: a.dataset.target,
+    title: a.parentElement.innerText,
+    choices: [],
+    allowCustom: false
+  };
+  switch (a.dataset.options) {
+    case "armor":
```
Comment (JFranklin3000) [inline module/helpers/traits.mjs:19]: change to armorTraits instead of armor.

File: templates/item/item-armor-sheet.hbs [hunk 2]
```diff
@@ -33,14 +33,17 @@
 
     {{!-- Traits --}}
     {{#> "systems/essence20/templates/item/parts/item-sheet-field.hbs" label='E20.ArmorTraits'}}
+    {{#*inline "item-field-edit"}}
+      <a class="trait-selector" style="flex-grow: 0;" data-options="armor" data-target="system.traits" title="{{localize 'E20.TraitSelector'}}">
```
Comment (JFranklin3000) [inline templates/item/item-armor-sheet.hbs:37]: if we do the first one change this to armorTraits as well.

File: module/helpers/traits.mjs (hunk 1 above)
Comment (phildominguez) [reply module/helpers/traits.mjs:19]: Holding off on this until armor upgrades happen

File: templates/item/item-armor-sheet.hbs (hunk 2 above)
Comment (phildominguez) [reply templates/item/item-armor-sheet.hbs:37]: Holding off on this until armor upgrades happen

**Review by JFranklin3000 — APPROVED** [review]

**Pull request merged**